bin/worker.js

```diff
@@ -1,5 +1,23 @@
+import { parseRequest } from '../lib/request.js'
+import { retrieveFile as defaultRetrieveFile } from '../lib/retrieval.js'
+
+// Hardcoded base URL for the file retrieval
+// In the future either user should supply the base URL
+// or worker should be retrieve database or chain
+const BASE_URL = 'yablu.net'
+
 export default {
-  async fetch (request, env, ctx) {
-    return await fetch(request)
+  async fetch (request, env, ctx, { retrieveFile = defaultRetrieveFile } = {}) {
+    if (request.method !== 'GET') {
+      return new Response('Method Not Allowed', { status: 405 })
+    }
+
+    const { proofSetId, pieceCid } = parseRequest(request)
+    if (!proofSetId || !pieceCid) {
+      return new Response('Missing required fields', { status: 400 })
+    }
+
+    // TODO: Record retrieval stats to D1 asynchronously (do not block response)
```

Comment:
https://developers.cloudflare.com/workers/runtime-apis/context/#waituntil

Reply:
I've created a separate issue for this #7